marklogic-data-hub/src/test/java/com/marklogic/hub/dhs/DeployUserAmpsTest.java

```diff
@@ -0,0 +1,185 @@
+        HubClient adminHubClient = runAsAdmin().newHubClient();
+
+        DocumentManager moduleDocManager = adminHubClient.getModulesClient().newDocumentManager();
+        DocumentWriteSet writeSet = moduleDocManager.newWriteSet();
+        StringHandle handle =  new StringHandle("'use strict';\n" +
+            "module.exports.getPiiData = module.amp(\n" +
+            "  function getPiiData() {\n" +
+            "    return cts.doc('/pii/test.json');\n" +
+            "  }\n" +
+            ");");
+        handle.setFormat(Format.TEXT);
+        writeSet.add("/ampTest/testModule.sjs", getPermissionsMetaDataHandle(), handle);
+        moduleDocManager.write(writeSet);
+
+        handle = new StringHandle("{\"ssn\": \"123-456-7890\"}");
+        handle.setFormat(Format.JSON);
+        DocumentMetadataHandle permissions = new DocumentMetadataHandle()
+            .withPermission("pii-reader", DocumentMetadataHandle.Capability.UPDATE,READ);
+        DocumentManager adminDocManager = adminHubClient.getStagingClient().newDocumentManager();
+        writeSet = adminDocManager.newWriteSet();
+        writeSet.add("/pii/test.json", permissions, handle);
+        adminDocManager.write(writeSet);
+
+        ServerEvaluationCall piiData = operatorHubClient.getStagingClient().newServerEval().javascript("const ampMod = require('/ampTest/testModule.sjs');\n" +
+            "ampMod.getPiiData();");
+
+        //query returns null in the absence of the amp
+        assertFalse(piiData.eval().hasNext());
+
+        ResourcesFragment amps;
+        try{
+            //POST request
+            writeAmpFileToProjectAndDeploy(HubConfig.DEFAULT_MODULES_DB_NAME, "pii-reader");
+            amps = new AmpManager(runAsDataHubSecurityAdmin().getManageClient()).getAsXml();
+
+            assertTrue(amps.resourceExists("getPiiData"));
+            JacksonHandle jacksonHandle = new JacksonHandle();
+            piiData.eval().next().get(jacksonHandle);
+            assertEquals("123-456-7890", jacksonHandle.get().get("ssn").asText(), "data-hub-operator should be " +
+                "able to read the document since the amp has been created");
+
+            //PUT request - shouldn't fail
+            writeAmpFileToProjectAndDeploy(HubConfig.DEFAULT_MODULES_DB_NAME,"data-hub-developer", "pii-reader");
+        }
+        finally {
+            runAsDataHubSecurityAdmin().getManageClient().delete("/manage/v2/amps/getPiiData?namespace=&document-uri=/ampTest/testModule.sjs&modules-database=data-hub-MODULES");
+            amps = new AmpManager(runAsDataHubSecurityAdmin().getManageClient()).getAsXml();
+            assertFalse(amps.resourceExists("getPiiData"));
+        }
+    }
+
+    @Test
+    void createAmpWithDelegatedUser(){
+        //create custom role having "data-hub-amp-manager" as "data-hub-security-admin"
+        ObjectNode roleNode = mapper.createObjectNode();
+        roleNode.put("role-name", "delegated-amp-manager");
```

Comment:
Can you start this role-name with "test-" - i.e. "test-delegated-amp-manager"? Just makes life easy in case it's left behind by a test - makes it obvious that it's a test role.